src/agent/src/rpc.rs

```diff
@@ -1341,6 +1398,41 @@ fn get_memory_info(block_size: bool, hotplug: bool) -> Result<(u64, bool)> {
     Ok((size, plug))
 }
 
+fn get_volume_capacity_stats(path: &str) -> Result<VolumeUsage> {
+    let mut usage = VolumeUsage::new();
+
+    let s = System::new();
+    for disk in s.disks() {
+        if disk.name().to_str().unwrap().eq(path) {
```

Comment:
I think we should return an error if `to_str()` fails.

Reply:
Updated. But I am very very new to Rust. Let me know if there is a better way to write this.